tests/codex/slots/testprover.nim

```diff
@@ -24,52 +24,69 @@ import ./backends/helpers
 
 suite "Test Prover":
   let
-    slotId = 1
     samples = 5
-    ecK = 3
-    ecM = 2
-    numDatasetBlocks = 8
     blockSize = DefaultBlockSize
     cellSize = DefaultCellSize
     repoTmp = TempLevelDb.new()
     metaTmp = TempLevelDb.new()
+    r1cs = "tests/circuits/fixtures/proof_main.r1cs"
+    wasm = "tests/circuits/fixtures/proof_main.wasm"
+    circomBackend = CircomCompat.init(r1cs, wasm)
+    challenge = 1234567.toF.toBytes.toArray32
 
   var
-    datasetBlocks: seq[bt.Block]
     store: BlockStore
-    manifest: Manifest
-    protected: Manifest
-    verifiable: Manifest
-    sampler: Poseidon2Sampler
+    prover: Prover
 
   setup:
     let
       repoDs = repoTmp.newDb()
       metaDs = metaTmp.newDb()
 
     store = RepoStore.new(repoDs, metaDs)
+    prover = Prover.new(store, circomBackend, samples)
 
-    (manifest, protected, verifiable) =
+  teardown:
+    await repoTmp.destroyDb()
+    await metaTmp.destroyDb()
+
+  test "Should sample and prove a slot":
+    let
+      (_, _, verifiable) =
         await createVerifiableManifest(
           store,
-          numDatasetBlocks,
-          ecK, ecM,
+          8, # number of blocks in the original dataset (before EC)
+          5, # ecK
+          3, # ecM
           blockSize,
           cellSize)
 
-  teardown:
```

Comment:
Why remove this?

Reply:
This is just a weird diff, the teardown is still there.